force/sobjects.go

```diff
@@ -112,6 +114,41 @@ func (forceApi *ForceApi) DeleteSObject(id string, in SObject) (err error) {
 	return
 }
 
+func idsFromURIs(uris []string) []string {
+	ids := make([]string, len(uris))
+	for i, uri := range uris {
+		parts := strings.Split(uri, "/")
+		ids[i] = parts[len(parts)-1]
+	}
+	return ids
+}
+
+func (forceApi *ForceApi) GetSFIDsByExternalId(apiName, externalKey, externalId string) ([]string, int, error) {
+	uri := fmt.Sprintf("%v/%v/%v", forceApi.apiSObjects[apiName].URLs[sObjectKey], externalKey, externalId)
+
+	params := url.Values{"fields": []string{"Id"}}
+	sobj := sobjects.BaseSObject{}
+
+	statusCode, err := forceApi.Get(uri, params, &sobj)
+	if err == nil {
+		return []string{sobj.Id}, statusCode, nil
+	}
+
+	if statusCode != 300 {
+		return []string{}, statusCode, err
```

Comment:
Should we return a nil slice instead of []string{}? The caller won't expect to be able to use this.

Reply:
I actually like returning an empty slice so that the caller can still iterate through the list. If it needs to know there was an err, it should check err for nil.

Reply:
You can actually iterate over a nil slice too - it will behave like an empty slice. The only difference is if returnedSlice != nil or returnedSlice == nil.

Reply:
This is a pretty minor nit though so either way is fine.

Reply:
Good point. Changed.